File: Checks/PWR015/example-omp.f90
! PWR015: Avoid copying unnecessary array elements to the GPU

subroutine example(A, B, sum)
  implicit none
  integer, intent(in) :: A(:), B(:)
  integer, intent(out) :: sum(:)
  integer :: i

  !$omp target parallel do default(none) shared(A, B, sum) &
  !$omp& map(to: a, b) map(from: sum)
  do i = 1, size(sum, 1) / 2
    sum(i) = A(i) + B(i)
  end do
  !$omp end target parallel do
end subroutine example

File: Checks/PWR015/example-omp.c
// PWR015: Avoid copying unnecessary array elements to the GPU

void example() {
  int A[100], B[100], sum[100];
  #pragma omp target map(to: A[0:100], B[0:100]) map(from: sum[0:100])
  #pragma omp parallel for private(i)
  for (int i = 0; i < 50; i++) {
    sum[i] = A[i] + B[i];
  }
}
